program/src/processor.rs

```diff
@@ -153,6 +153,106 @@ impl<'a> RemoveOracleContext<'a> {
     }
 }
 
+struct AddRequesterContext<'a> {
+    rent: Rent,
+    aggregator: &'a AccountInfo<'a>,
+    aggregator_owner: &'a AccountInfo<'a>, // signed
+    requester: &'a AccountInfo<'a>,
+    requester_owner: &'a AccountInfo<'a>,
+
+    description: [u8; 32],
+}
+
+impl<'a> AddRequesterContext<'a> {
+    fn process(&self) -> ProgramResult {
+        let aggregator = Aggregator::load_initialized(self.aggregator)?;
+        msg!("loaded aggregator");
+        aggregator.authorize(self.aggregator_owner)?;
+
+        let mut requester = Oracle::init_uninitialized(self.requester)?;
```

Comment:
why is the requester using the same struct as Oracle?

Reply:
i think this is a security issue. If a requester has exactly the same struct as an Oracle, the requester would be able to submit value as an oracle.

Note also that if you create a Requester struct, but its binary representation can deserialize as an Oracle, you'd have the same security hole.